Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -251,5 +263,516 @@ namespace AZ
+            {
+                const AZStd::string label =
+                    AZStd::string::format("%s/ %s", block.m_groupRegionName->m_groupName, block.m_groupRegionName->m_regionName);
+                const float textWidth = ImGui::CalcTextSize(label.c_str()).x;
+
+                if (regionPixelWidth < textWidth) // Not enough space in the block to draw the whole name, draw clipped text.
+                {
+                    // clipRect appears to only clip when a character is fully outside of its bounds which can lead to overflow
+                    // for now subtract the width of a character
+                    const ImVec4 clipRect = { startPoint.x, startPoint.y, endPoint.x - maxCharWidth, endPoint.y };
+                    const float fontSize = ImGui::GetFont()->FontSize;
+
+                    ImGui::GetFont()->RenderText(drawList, fontSize, startPoint, IM_COL32_WHITE, clipRect, label.c_str(), 0);
+                }
+                else // We have enough space to draw the entire label, draw and center text.
+                {
+                    const float remainingWidth = regionPixelWidth - textWidth;
+                    const float offset = remainingWidth * .5;
+
+                    drawList->AddText({ startPoint.x + offset, startPoint.y }, IM_COL32_WHITE, label.c_str());
+                }
+            }
+
+            // Tooltip and block highlighting
+            if (ImGui::IsMouseHoveringRect(startPoint, endPoint) && ImGui::IsWindowHovered())
+            {
+                // Open function statistics map on click
+                if (ImGui::IsMouseClicked(ImGuiMouseButton_Left))
+                {
+                    const GroupRegionName* key = block.m_groupRegionName;
+                    m_regionStatisticsMap[key].m_draw = true;
+                }
+
+                // Hovering outline
+                drawList->AddRect(startPoint, endPoint, ImGui::GetColorU32({ 1, 1, 1, 1 }), 0.0, 0, 1.5);
+
+                ImGui::BeginTooltip();
+                ImGui::Text("%s::%s", block.m_groupRegionName->m_groupName, block.m_groupRegionName->m_regionName);
+                ImGui::Text("Execution time: %.3f ms", CpuProfilerImGuiHelper::TicksToMs(block.m_endTick - block.m_startTick));
+                ImGui::Text("%lld => %lld", block.m_startTick, block.m_endTick);
+                ImGui::EndTooltip();
+            }
+        }
+
+        inline ImU32 ImGuiCpuProfiler::GetBlockColor(const TimeRegion& block)
+        {
+            // Use the GroupRegionName pointer a key into the cache, equal regions will have equal pointers
+            const GroupRegionName* key = block.m_groupRegionName;
+            if (m_regionColorMap.contains(key)) // Cache hit
+            {
+                return ImGui::GetColorU32(m_regionColorMap[key]);
+            }
+
+            // Cache miss, generate a new random color
+            std::mt19937 mt(m_rd());
+            std::uniform_real_distribution<float> dis(.1, .9);
```

Comment:
I think we should avoid using any std:: functions or types directly. 
We have some random functions in AzCore: \Code\Framework\AzCore\AzCore\Math\Random.h